src/lando/main/tests/test_git.py

```diff
@@ -133,6 +136,46 @@ def test_GitSCM_clean_repo(
     ), f"strip_non_public_commits not honoured for {new_file}"
 
 
+def test_GitSCM_apply_get_patch(git_repo: Path, git_patch: Callable):
+    scm = GitSCM(str(git_repo))
+
+    patch = git_patch()
+
+    ph = GitPatchHelper(io.StringIO(patch))
+
+    author_name, author_email = ph.parse_author_information()
+    author = f"{author_name} <{author_email}>"
+    scm.apply_patch(
+        ph.get_diff(), ph.get_commit_description(), author, ph.get_timestamp()
+    )
+
+    commit = scm.describe_commit()
+
+    expected_patch = patch
+    new_patch = scm.get_patch(commit.hash)
+
+    # Trim first line from both patches, as they will contain a `From` line
+    # with a different base commit.
+    trim_from_re = r"^From [^\n]+\n"
+    expected_patch = re.sub(trim_from_re, "", expected_patch, count=1)
+    new_patch = re.sub(trim_from_re, "", new_patch, count=1)
+
+    # The git version used when we created our fixture may no longer match the one on
+    # the system running the test.
+    unify_git_version_re = r"\d+(\.\d+)+$"
+    expected_patch = re.sub(unify_git_version_re, "GIT.VERS.ION", expected_patch)
+    new_patch = re.sub(unify_git_version_re, "GIT.VERS.ION", new_patch)
```

Comment:
nit: I think it would be cleaner to explicitly ignore this, rather than modifying it in the expected output. This may mean that we would be going through the patch a little differently when comparing, rather than what you have on line 176.